common/src/test/scala/com/nvidia/spark/rapids/ThreadFactoryBuilderTest.scala

```diff
@@ -26,34 +26,43 @@ class ThreadFactoryBuilderTest extends FunSuite {
     var pool = Executors.newFixedThreadPool(2,
       new ThreadFactoryBuilder().setNameFormat("Thread %s").setDaemon(true).build())
 
-    pool.submit(new Runnable {
-      override def run(): Unit = {
+    var ret = pool.submit(new Callable[String] {
+      override def call(): String = {
         assert(Thread.currentThread().isDaemon)
-        assert(Thread.currentThread().getName == "Thread 0")
+        assert(Thread.currentThread().getName() == "Thread 0")
+        ""
       }
     })
-
-    pool.submit(new Runnable {
-      override def run(): Unit = {
+    // waits and retrieves the result, if above asserts failed, will get execution exception
+    ret.get()
+    ret = pool.submit(new Callable[String] {
+      override def call(): String = {
         assert(Thread.currentThread().isDaemon)
-        assert(Thread.currentThread().getName == "Thread 1")
+        assert(Thread.currentThread().getName() == "Thread 1")
+        ""
       }
     })
+    ret.get()
+    pool.shutdown()
```

Comment:
`pool` won't be shutdown if there's an exception.  This needs to be in a `finally` block.  Same applies to the shutdown call below.